examples/docs_snippets/docs_snippets/guides/dagster/ml_ops/ml_ops.py

```diff
@@ -0,0 +1,139 @@
+import pandas as pd
+import requests
+import numpy as np
+from dagster import (
+    MetadataValue,
+    Output,
+    asset,
+    Definitions,
+    DynamicPartitionsDefinition,
+    Output,
+    FreshnessPolicy,
+)
+import datetime
+from sklearn.model_selection import train_test_split
+from sklearn.feature_extraction.text import CountVectorizer
+from sklearn import linear_model
+
+
+@asset
+def previous_item():
+    previous_item = requests.get(
+        f"https://hacker-news.firebaseio.com/v0/maxitem.json"
+    ).json()
+    return previous_item
+
+
+## data_ingestion_start
+
+
+@asset(freshness_policy=FreshnessPolicy(maximum_lag_minutes=10))
+def hackernews_stories(previous_item):
+    """Get the max ID number from hacker news"""
+
+    latest_item = requests.get(
+        f"https://hacker-news.firebaseio.com/v0/maxitem.json"
+    ).json()
+
+    """Get items based on story ids from the HackerNews items endpoint"""
+    results = []
+    scope = range(previous_item, latest_item)
+    for item_id in scope:
+        item = requests.get(
+            f"https://hacker-news.firebaseio.com/v0/item/{item_id}.json"
+        ).json()
+        results.append(item)
+
+    df = pd.DataFrame(results)
+    if len(df) > 0:
+        df = df[df.type == "story"]
+        df = df[~df.title.isna()]
+
+    return df, latest_item
+
+
+## data_ingestion_end
+
+## test_train_split_start
+
+
+@asset
+def test_train_split(hackernews_stories):
+    hackernews_stories, max_item = hackernews_stories
+    X = hackernews_stories.title
+    y = hackernews_stories.descendants
+    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2)
+    return X_train, X_test, y_train, y_test
+
+
+## test_train_split_end
+
+
+## vectorizer_start
+@asset
+def transform_train(test_train_split):
+    X_train, X_test, y_train, y_test = test_train_split
+    vectorizer = TfidfVectorizer()
+    transformed_X_train = vectorizer.fit_transform(X_train)
+    transformed_X_train = transformed_X_train.toarray()
+    y_train = y_train.fillna(0)
+    transformed_y_train = np.array(y_train)
+
+    return vectorizer, transformed_X_train, transformed_y_train
```

Comment:
thought would be that if you wanted to take a new dataset and transform it to do predictions you would want to use it

Reply:
Ah, I see. It seems like we basically need it in any place we want to do inference? If that's the case, what do you think about combining it with the predictive model itself into a class called something like `StoryTitleCommentsPredictor`? Then, anywhere we need to do inference, we can just depend on that single asset.

Also, I know it adds some complexity, but I think it could be worth adding an inference step to the pipeline? Maybe in a separate Python file?

Reply:
added inference asset - do we want it to predict or score a new dataset?